Gems/Atom/Tools/AtomToolsFramework/Code/Include/AtomToolsFramework/EntityPreviewViewport/EntityPreviewViewportScene.h

```diff
@@ -27,20 +27,35 @@ namespace AtomToolsFramework
             RenderViewportWidget* widget,
             AZStd::shared_ptr<AzFramework::EntityContext> entityContext,
             const AZStd::string& sceneName = "EntityPreviewViewportScene",
-            const AZStd::string& pipelineAssetPath = "passes/MainRenderPipeline.azasset");
+            const AZStd::string& defaultRenderPipelineAssetPath = "passes/mainrenderpipeline.azasset");
         ~EntityPreviewViewportScene();
 
+        bool ActivateRenderPipeline(const AZStd::string& pipelineAssetPath);
+        bool ActivateRenderPipeline(const AZ::Data::AssetId& pipelineAssetId);
+
         //! Returns a pointer to the scene used for rendering the viewport content
         AZ::RPI::ScenePtr GetScene() const;
 
         //! Returns a pointer to the pipeline used for rendering the viewport content
         AZ::RPI::RenderPipelinePtr GetPipeline() const;
 
+        //! Returns the AssetId of the active render pipeline 
+        AZ::Data::AssetId GetPipelineAssetId() const;
+
     private:
+        using RenderPipelineMap = AZStd::unordered_map<AZ::Data::AssetId, AZ::RPI::RenderPipelinePtr>;
+
+        RenderPipelineMap::iterator AddRenderPipeline(const AZ::Data::AssetId& pipelineAssetId);
+
         const AZ::Crc32 m_toolId = {};
         AZ::RPI::ScenePtr m_scene;
         AZStd::shared_ptr<AzFramework::Scene> m_frameworkScene;
-        AZ::RPI::RenderPipelinePtr m_renderPipeline;
+
+        RenderPipelineMap m_renderPipelines;
+        RenderPipelineMap::iterator m_activeRenderPipeline;
```

Comment:
It is probably best to avoid storing iterators as members because they can be invalidated whenever the container is mutated. Instead, you could just copy the pair or store the active asset ID and pipeline pointer in their own members.

Reply:
In this case it's safe because every time the map is mutated we update m_activeRenderPipeline. But I'll change it anyway to get rid of the "funny smell".